sdk/clientcore/annotation-processor/src/test/java/io/clientcore/annotation/processor/models/TemplateInputTest.java

```diff
@@ -63,9 +65,7 @@ void addImportIgnoresEmptyImport() {
     @Test
     void addImportTypeMirrorAddsValidImport() {
         TemplateInput templateInput = new TemplateInput();
-        DeclaredType declaredType = mock(DeclaredType.class);
-        when(declaredType.toString()).thenReturn("java.util.Map");
-        when(declaredType.getKind()).thenReturn(TypeKind.DECLARED);
+        DeclaredType declaredType = new MockDeclaredType(TypeKind.DECLARED, "java.util.Map");
```

Comment:
Just curious why we moved from mock testing to actual Mock classes?

Reply:
Mockito is something we really only want to use when all else fails. It does a lot of reflection magic, and we're stuck on the Java 8 release that is years old now which we've found issues with due to reflection changes in newer versions of Java.